test/unittests/execute_floating_point_test.cpp

```diff
@@ -137,8 +137,8 @@ class TestValues
             std::array<T, ps.size() * 2 + 1> a;
 
             auto it = std::begin(a);
-            it = std::transform(std::reverse_iterator{std::end(ps)},
-                std::reverse_iterator{std::begin(ps)}, it, std::negate<T>{});
+            it = std::transform(std::make_reverse_iterator(std::end(ps)),
```

Comment:
Does this resolve some warning?

Reply:
There is warning that `reverse_iterator` is may not to designed for CTAD (as it does not have any explicit CTAD hints). I think it is ok to change it here as this is the only case for this warning. And I'm also considering rewriting these test arrays to constexpr version where STL algorithms are going to be less useful.